crates/test-utils/src/deploy_helpers.rs

```diff
@@ -831,6 +746,71 @@ impl KakarotTestEnvironmentContext {
         Self { sequencer, kakarot_client, kakarot, kakarot_contract, evm_contracts }
     }
 
+    /// Initializes the Kakarot test environment and deploys the Kakarot system and EVM contracts.
+    async fn initialize_without_dumped_state(
+        sequencer: TestSequencer,
+        starknet_provider: Arc<JsonRpcClient<HttpTransport>>,
+    ) -> Self {
+        let starknet_account = sequencer.account();
+
+        // Define the expected funded amount for the Kakarot system
+        let expected_funded_amount = FieldElement::from_dec_str("1000000000000000000").unwrap();
+
+        // Deploy the Kakarot system
+        let kakarot = deploy_kakarot_system(&sequencer, EOA_WALLET.clone(), expected_funded_amount).await;
+
+        let starknet_deployer_account = deploy_deployer_account(starknet_provider.clone(), &starknet_account).await;
+
+        // Create a Kakarot client
+        let kakarot_client = KakarotClient::new(
+            StarknetClientConfig::new(
+                Network::JsonRpcProvider(sequencer.url()),
+                kakarot.kakarot_address,
+                kakarot.proxy_class_hash,
+            ),
+            starknet_provider,
+            starknet_deployer_account,
+        );
+
+        let kakarot_contract =
```

Comment:
it's surprising to see that "kakarot system" actually doesn't have the only "real" kakarot, ie the kakarot contract

Reply:
Would this be related to https://github.com/kkrt-labs/kakarot-rpc/issues/527? I think if we move to getting class hashes from the contact, we can simplify `KakarotContract`